src/test/java/org/jenkinsci/plugins/oic/PluginTest.java

```diff
@@ -353,11 +375,11 @@ public void testRefreshTokenAndTokenExpiration_withoutRefreshToken() throws Exce
         mockUserInfoWithTestGroups();
         browseLoginPage();
         assertTestUser();
-
         // expired token not refreshed
         expire();
-        webClient.assertFails(jenkins.getSecurityRealm().getLoginUrl(), 302);
-
+        // use an actual HttpClient to make checking redirects easier
+        HttpResponse<String> rsp = getPageWithGet("/manage");
```

Comment:
Is `JenkinsRule#createWebClient` not enough for this?

Reply:
not the way the test was originally written - AFAICT `JenkinsRule.WebClient` follows http redirects, which was causing the test to fail.  The test  was passing before because `JenkinsRule.WebClient` was erroring out due to the redirect loop.  I used the `HttpClient` because you can tell it not to follow redirects.

Reply:
@amuniz just to be certain, i took a look at the original test and confirmed that it does indeed suffer from the same redirect loop
![image](https://github.com/user-attachments/assets/6fe8b4d1-d59e-41e3-8145-59e883c60cf7)